src/html-behavior.js

```diff
@@ -401,6 +405,33 @@ export class HtmlBehaviorResource {
     return controller;
   }
 
+  /**
+   * Allow a bindable property on custom element to register how to reflect prop value to attribute
+   * @param {string} propertyName 
+   * @param {{(element: Element, name: string, newVal, oldVal) => any}} instruction A function with suitable parameters to react for setting attribute on the element
+   */
+  _registerReflection(propertyName, instruction) {
+    let reflections = this.reflections || (this.reflections = {});
+    if (propertyName in reflections) {
+      throw new Error(`Reflection for ${propertyName} was already registered`);
+    }
+    if (typeof instruction !== 'function') {
+      throw new Error('Invalid reflection instruction');
+    }
+    reflections[propertyName] = instruction;
+  }
+  
+  /**
+   * @param {Element} element 
```

Comment:
Why use Typescript-like type annotations in `bindable-property.js` and JSDoc comments here?